pytorch_lightning/loggers/neptune.py

```diff
@@ -46,355 +94,557 @@ class NeptuneLogger(LightningLoggerBase):
+                + msg_suffix
+            )
+        else:
+            raise ValueError("The function you've used is deprecated.\n" + msg_suffix)
 
-        Args:
-            metric_name: The name of log, i.e. mse, loss, accuracy.
-            metric_value: The value of the log (data-point).
-            step: Step number at which the metrics should be recorded, must be strictly increasing
-        """
-        if is_tensor(metric_value):
+    @rank_zero_only
+    def log_metric(self, metric_name: str, metric_value: Union[torch.Tensor, float, str], step: Optional[int] = None):
+        key = f"{self._prefix}/{metric_name}"
+        self._signal_deprecated_api_usage("log_metric", f"logger.run['{key}'].log(42)")
+        if torch.is_tensor(metric_value):
             metric_value = metric_value.cpu().detach()
 
-        if step is None:
-            self.experiment.log_metric(metric_name, metric_value)
-        else:
-            self.experiment.log_metric(metric_name, x=step, y=metric_value)
+        self.run[key].log(metric_value, step=step)
 
     @rank_zero_only
     def log_text(self, log_name: str, text: str, step: Optional[int] = None) -> None:
-        """Log text data in Neptune experiments.
-
-        Args:
-            log_name: The name of log, i.e. mse, my_text_data, timing_info.
-            text: The value of the log (data-point).
-            step: Step number at which the metrics should be recorded, must be strictly increasing
-        """
-        if step is None:
-            self.experiment.log_text(log_name, text)
-        else:
-            self.experiment.log_text(log_name, x=step, y=text)
+        key = f"{self._prefix}/{log_name}"
+        self._signal_deprecated_api_usage("log_text", f"logger.run['{key}].log('text')")
+        self.run[key].log(str(text), step=step)
 
     @rank_zero_only
     def log_image(self, log_name: str, image: Union[str, Any], step: Optional[int] = None) -> None:
-        """Log image data in Neptune experiment.
-
-        Args:
-            log_name: The name of log, i.e. bboxes, visualisations, sample_images.
-            image: The value of the log (data-point).
-                Can be one of the following types: PIL image, `matplotlib.figure.Figure`,
-                path to image file (str)
-            step: Step number at which the metrics should be recorded, must be strictly increasing
-        """
-        if step is None:
-            self.experiment.log_image(log_name, image)
-        else:
-            self.experiment.log_image(log_name, x=step, y=image)
+        key = f"{self._prefix}/{log_name}"
+        self._signal_deprecated_api_usage("log_image", f"logger.run['{key}'].log(File('path_to_image'))")
+        if isinstance(image, str):
+            # if `img` is path to file, convert it to file object
+            img = NeptuneFile(image)
+        self.run[key].log(img, step=step)
```

Comment:
```suggestion
        if isinstance(image, str):
            # if `img` is path to file, convert it to file object
            image = NeptuneFile(image)
        self.run[key].log(image, step=step)
```
local variable 'img' referenced before assignment